install.d/50-dracut.install

```diff
@@ -2,12 +2,12 @@
 
 COMMAND="$1"
 KERNEL_VERSION="$2"
+#shellcheck disable=SC2034
 BOOT_DIR_ABS="$3"
 KERNEL_IMAGE="$4"
 
-# If KERNEL_INSTALL_MACHINE_ID is defined but empty, BOOT_DIR_ABS is a fake directory.
-# So, let's skip to create initrd.
-if ! [[ ${KERNEL_INSTALL_MACHINE_ID-x} ]]; then
+# If the initrd was provided on the kernel command line, we shouldn't generate our own.
+if [[ "$COMMAND" != "add" || "$#" -gt 4 ]]; then
```

Comment:
With this, the support for `remove` is dropped. I can't support this.

Reply:
Removal is handled by other kernel-install plugins. 

This plugin should only generate an initrd or uki.efi and put it on the temporary staging area (`KERNEL_INSTALL_STAGING_AREA`), it does not install the inird or uki.efi to /boot or ESP, because where we want it depends on the `KERNEL_INSTALL_LAYOUT`. Since we have ` BOOT_DIR_ABS="$KERNEL_INSTALL_STAGING_AREA"` and the staging area is unset when removing the kernel the `remove` case never actually did anything.

Reply:
Oh, ok. I didn't know it's obsoleted (if it was ever used).

Reply:
I suppose it was used before we changed the plugin to respect and use the `KERNEL_INSTALL_STAGING_AREA`.

Reply:
> Removal is handled by other kernel-install plugins.
> 

Is it still the grub plugin? If so, that's a thing that should be fixed.

Reply:
AFAIK there is no upstream grub plugin, neither in grub nor in systemd. We have one [downstream in Gentoo](https://github.com/projg2/installkernel-gentoo/blob/master/hooks/systemd/90-compat.install) that handles installing (from staging area) and removing the initramfs when the kernel is installed or removed. This mirrors what the upstream systemd-boot plugin does, it also handles installing (from staging area) and removing initramfs along with the kernel.